sticht/rollbacks/sources/splunk.py

```diff
@@ -24,26 +25,33 @@ def __init__(
         label: str,
         query: str,
         on_failure_callback: Callable[['MetricWatcher'], None],
-        splunk_host: str,
-        splunk_port: int,
-        credentials_callback: Callable[[], Tuple[str, str]],
+        auth_callback: Callable[[], SplunkAuth],
     ) -> None:
         super().__init__(label, on_failure_callback)
         self._query = query
         # TODO: should we share a global version of this so that we're
         # not logging-in a million times?
         self._splunk: Optional[splunklib.client.Service] = None
-        self._credentials_callback = credentials_callback
-        self._splunk_host = splunk_host
-        self._splunk_port = splunk_port
+        self._credentials_callback = auth_callback
+
+    def __eq__(self, other: object) -> bool:
```

Comment:
needed this so that i could test from_config - we probably won't normally need to compare these :p